cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1692,7 +1692,16 @@ private AppFabricServiceStatus stop(Id.Program identifier, ProgramType type) {
     try {
       Preconditions.checkNotNull(runtimeInfo, UserMessages.getMessage(UserErrors.RUNTIME_INFO_NOT_FOUND));
       ProgramController controller = runtimeInfo.getController();
-      controller.stop().get();
+      if (type == ProgramType.WORKFLOW) {
+        try {
+          controller.stop().get();
+        } catch (IllegalArgumentException ex) {
+          // Stopping workflow throws IllegalArgumentException
```

Comment:
We need to throw an exception so that it can be logged as `ERROR` in the metadata store. Should it be `InterruptedException` instead?